kernel/packages/unity-interface/fetchENSOwnerProfile.ts

```diff
@@ -0,0 +1,19 @@
+import { isAddress } from 'eth-connect'
+import { ethereumConfigurations } from 'config'
+import { fetchENSOwnersContains, getAppNetwork } from 'shared/web3'
+import { Profile, ProfileType } from 'shared/profiles/types'
+import { ProfileAsPromise } from 'shared/profiles/ProfileAsPromise'
+
+export function fetchENSOwnerProfile(name: string, maxResults: number = 1): Promise<Profile[]> {
```

Comment:
I'd think making this an async function would allow for easier to read code. Then you can use await below, and you wouldn't need to call then or Promise.resolve. I think.